python/lsst/meas/base/plugins.py

```diff
@@ -71,6 +73,72 @@
                     TransformClass=bl.ApertureFluxTransform, executionOrder=BasePlugin.FLUX_ORDER)
 
 # --- Single-Frame Measurement Plugins ---
+class SingleFrameFocalPlaneConfig(SingleFramePluginConfig):
+    pass
+
+@register("base_FocalPlane")
+class SingleFrameFocalPlanePlugin(SingleFramePlugin):
+    '''
+    Algorithm to calculate position on focal plane
+    '''
+
+    ConfigClass = SingleFrameFocalPlaneConfig
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.APCORR_ORDER
+
+    def __init__(self, config, name, schema, metadata):
+        SingleFramePlugin.__init__(self, config, name, schema, metadata)
+        self.focalValue = lsst.afw.table.Point2DKey.addFields(schema, name, "Position on the focal plane",
+                "mm")
+        self.focalFlag = schema.addField(name + "_flag", type="Flag", doc="Set to 1 for any fatal Failure")
+
+    def measure(self, measRecord, exposure):
+        center = measRecord.getCentroid()
+        det = exposure.getDetector()
+        if not det:
+            print('in first')
+            fp = lsst.afw.geom.Point2D(numpy.nan, numpy.nan)
+        else:
+            posInPix = det.makeCameraPoint(center, lsst.afw.cameraGeom.PIXELS)
+            fp = det.transform(posInPix, lsst.afw.cameraGeom.FOCAL_PLANE).getPoint()
+        measRecord.set(self.focalValue, fp)
+
+    def fail(self, measRecord, error=None):
+        measRecord.set(self.focalFlag, True)
+
+class SingleFrameJacobianConfig(SingleFramePluginConfig):
+    pixelScale = lsst.pex.config.Field(dtype=float, default=0.5, doc="Nominal pixel size (arcsec)")
```

Comment:
The idea is to have a single nominal pixel size for all CCDs.  If you use `exposure.getWcs().getPixelScale()`, then different CCDs can have different nominal pixel sizes, which means the Jacobian will be discontinuous over CCDs; but the whole idea is to make things continuous over the focal plane.